patches/components-sync_device_info-device_info_sync_bridge.cc.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/components/sync_device_info/device_info_sync_bridge.cc b/components/sync_device_info/device_info_sync_bridge.cc
+index a809fd09ea04d28b7293d1554f29f0c83dd648a3..c0066b8ab45e9956a7d094d9bd9ddd45d5e674c3 100644
+--- a/components/sync_device_info/device_info_sync_bridge.cc
++++ b/components/sync_device_info/device_info_sync_bridge.cc
+@@ -254,7 +254,7 @@ std::string DeviceInfoSyncBridge::GetStorageKey(const EntityData& entity_data) {
+ void DeviceInfoSyncBridge::ApplyStopSyncChanges(
+     std::unique_ptr<MetadataChangeList> delete_metadata_change_list) {
+   if (!delete_metadata_change_list) {
+-    return;
```

Comment:
Since this isn't really intuitive can you make it a separate commit so we can attach some context to it without adding comments to the patch?

Reply:
actually just explaining some of these in the PR would be good enough

Reply:
This is to workaround this DCHECK https://cs.chromium.org/chromium/src/components/sync_device_info/device_info_sync_bridge.cc?l=433&rcl=27bdc9521f34a4525005433b19c181c3e571a616
DCHECK failed when restart browser after sync chain established
We did calling `ApplyStopSyncChanges` when browser shutting down, just `delete_metadata_change_list` is null though but the following code doesn't reference it. But if we keep this workaround, we have to monitor the future change of this function in case it get referenced
I didn't have enough time to dig in, we rely on `DEVICE_INFO` type to keep scheduler running for us to fetch our brave device record. Maybe properly faking `DeviceInfoSpecifics` will fix it.

Reply:
https://github.com/brave/brave-browser/issues/5728 follow up issued created